### PR TITLE
adding Stored and Concat quantifier to fix Swap

### LINKED ISSUE #57: add VerifyConcatenate and IsStored atomic predicates
## Description

* VerifyConcatenate(bytes, bytes_a, bytes_b) for confirmation signature.
* separate IsStored(root, block_number) and verifyInclusion(leaf, range, token, root, proof).

VerifyConcatenate verified bytes is concatenated of bytes_a and bytes_b.
IsStored verifies `root` is stored in `block_number`.

### Task List

* [x] add IsStored quantifier
* [x] add VerifyConcat quantifier
* [x] add Stored quantifier
* [x] add Concat quantifier
* [x] add Hash quantifier
* [x] fix swap example

### PR DESCRIPTION
**Pull request by syuhei176**
## Description

I’ll apply 2 changes to ovm for making swap predicate.
* VerifyConcatenate(bytes, bytes_a, bytes_b) for confirmation signature.
* separate IsStored(root, block_number) and verifyInclusion(leaf, range, token, root, proof).

VerifyConcatenate verified bytes is concatenated of bytes_a and bytes_b.
IsStored verifies `root` is stored in `block_number`.

### Task List

* [x] add Stored quantifier
* [x] add Concat quantifier
* [x] add Hash quantifier
* [x] fix swap example

Close #57

### CODE REVIEW

File: packages/transpiler/src/utils.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export function isAtomicPredicate(predicate: string) {
     predicate == 'IsValidSignature' ||
     predicate == 'IsContained' ||
     predicate == 'VerifyInclusion' ||
+    predicate == 'VerifyConcatenate' ||
+    predicate == 'VerifyRoot' ||
```
Comment (syuhei176) [inline packages/transpiler/src/utils.ts]: we should consider about the namings

**Comment by tkmct** [comment]
what about naming it `IsConcatenatedWith`?
```
const a = Bytes(...)
const b = Bytes(...)
const c = Bytes.concatenate(a, b)

c.isConcatenatedWith(a, b)
```

**Comment by syuhei176** [comment]
> what about naming it `IsConcatenatedWith`?
> 
> ```
> const a = Bytes(...)
> const b = Bytes(...)
> const c = Bytes.concatenate(a, b)
> 
> c.isConcatenatedWith(a, b)
> ```

make sense. also we can have `leaf.IsIncluded(token, range, root, proof)`.

File: examples/testcases/operators/TestAnd.sol [hunk 2]
```diff
@@ -21,6 +21,8 @@ contract AndTest {
     address IsValidSignature = address(0x0000000000000000000000000000000000000000);
     address IsContained = address(0x0000000000000000000000000000000000000000);
     address VerifyInclusion = address(0x0000000000000000000000000000000000000000);
+    address VerifyRoot = address(0x0000000000000000000000000000000000000000);
+    address VerifyConcatenate = address(0x0000000000000000000000000000000000000000);
```
Comment (tkmct) [inline examples/testcases/operators/TestAnd.sol]: Don't you change these name as well?

Comment (syuhei176) [reply examples/testcases/operators/TestAnd.sol]: thanks! fixed

**Pull request closed without merging**